[BrM] Gift of the Ox

emallson opened the pull request:
![screenshot from 2018-12-21 11-44-46](https://user-images.githubusercontent.com/4909458/50353313-d525ff00-0515-11e9-9271-d0bc4ebd0e45.png)

Also includes code to estimate the value of agility / mastery to the heal, which is not used yet but is being used in an in-progress stat weights PR.

sMteX commented on src/parser/monk/brewmaster/normalizers/ExpelHarm.js:
```diff
@@ -0,0 +1,43 @@
+      // find the start of run of orbs consumed
+      // safe --- game won't let you cast EH with 0 orbs
+      let i = idx;
+      while(events[i-1].timestamp === event.timestamp) {
```
I think some kind of buffer would be helpful here. In this log for example https://wowanalyzer.com/report/t2ZK6pQvgLGqj7by/1-Mythic+MOTHER+-+Kill+(4:19)/14-Mantyke/events, I don't think the heal events from EH happen exactly at the same timestamp, e.g. here I think at least the last 2 of the 3 heals were triggered from EH, yet they're not on the same timestamp
![image](https://user-images.githubusercontent.com/5068584/50423325-24409c80-0854-11e9-8df0-3fda11babc26.png)

emallson replied:
good catch, i hadn't seen any so i didn't implement it but it is handled now :+1:

sMteX commented on src/parser/monk/brewmaster/modules/spells/GiftOfTheOx.js:
```diff
@@ -0,0 +1,107 @@
+    //           = Heal * (BonusAgi / (6 WDPS + BonusAgi + BaseAgi))
+    //
+    // and similar for bonus WDPS healing and base agi healing
+    const denom = (6 * this._wdps + this.stats.currentAgilityRating);
```
I don't really understand the math here, why and how, so I'm just gonna trust you on this 😅  If it's wrong, it's wrong just for BMs 🤷‍♀️

emallson commented:
@sMteX the bare `EventFilter` has been removed and replaced with an event imported from the gotox normalizer (which was also changed `tick -> orb-generated` so that I can just filter on that).
